### ISSUE TITLE
Funding recipient on deposit to be address(this)

### ISSUE THREAD

**Issue by petrovska-petro** [Citadel-DAO/citadel-contracts#18]
Looks like on deposit it goes to `saleRecipient` already. [ref](https://github.com/Citadel-DAO/citadel-contracts/blob/main/src/Funding.sol#L179), while i assume that the intention is to batch it thru this [method](https://github.com/Citadel-DAO/citadel-contracts/blob/main/src/Funding.sol#L315)

**Comment by dapp-whisperer** [comment]
I agree, I'm inclined to optimize this cost for the user but having the assets sit in the funding contract until swept. Then we can open up claimAssetToTreasury() to a keeper.

we could have the user approve the treasury vault but that's lol as the logic for the treasury vault is open-ended (multisig).

**Comment by GalloDaSballo** [comment]
So long as the only way to move `asset` is via governance and via the specific path that allows the `treasury` to receive funds (security concerns)

The modification should save between 5k and 10k gas so it's massive gas savings for all users
